qutebrowser/browser/webengine/webenginetab.py

```diff
@@ -1237,6 +1243,133 @@ def _inject_site_specific_quirks(self):
+        self._tab.abort_questions.emit()
+        self._request = None
+
+    def _ux_request_failed(self):
+        assert self._request is not None
+        log.webview.debug("Webauth request failed for "
+                          f"{self._request.relyingPartyId()}: "
+                          f"{self._request.requestFailureReason()}")
+
+        reason = {
+            QWebEngineWebAuthUxRequest.RequestFailureReason.Timeout:
+                "The request timeout out.",
+            QWebEngineWebAuthUxRequest.RequestFailureReason.KeyNotRegistered:
+                "The device is not registered.",
+            QWebEngineWebAuthUxRequest.RequestFailureReason.KeyAlreadyRegistered:
+                "You already registered this device.",
+            QWebEngineWebAuthUxRequest.RequestFailureReason.SoftPinBlock:
+                "The device is locked because the wrong PIN was entered too many times.",
+            QWebEngineWebAuthUxRequest.RequestFailureReason.HardPinBlock:
+                "The device is locked because the wrong PIN was entered too many times.",
+            QWebEngineWebAuthUxRequest.RequestFailureReason.AuthenticatorRemovedDuringPinEntry:
+                "The device was removed during verification. Please reinsert and try again",
+            QWebEngineWebAuthUxRequest.RequestFailureReason.AuthenticatorMissingResidentKeys:
+                "The device does not have resident key support.",
+            QWebEngineWebAuthUxRequest.RequestFailureReason.AuthenticatorMissingUserVerification:
+                "The device is missing user verification.",
+            QWebEngineWebAuthUxRequest.RequestFailureReason.AuthenticatorMissingLargeBlob:
+                "The device s missing Large Blob support.",
+            QWebEngineWebAuthUxRequest.RequestFailureReason.NoCommonAlgorithms:
+                "The device is missing Large Blob support.",
+            QWebEngineWebAuthUxRequest.RequestFailureReason.StorageFull:
+                "The storage on the device is full.",
+            QWebEngineWebAuthUxRequest.RequestFailureReason.UserConsentDenied:
+                "User consent was denied.",
+            QWebEngineWebAuthUxRequest.RequestFailureReason.WinUserCancelled:
+                "User cancelled the request."
+        }[self._request.requestFailureReason()]
+        message.info(f"User verification failed: {reason}")
+        self._request = None
+
+    def _verification_required(self, url):
+        """Ask a prompt for a webuth user verification request."""
+        return message.ask(
+            title=f"User Verification for {url}",
+            text="Please enter the PIN for your device:",
+            mode=usertypes.PromptMode.pwd,
+            abort_on=[self._tab.abort_questions])
+
+    def _select_account(self, url, usernames):
+        """Ask a prompt for a webuth account selection."""
+        text = "Please select the your account:<br>" \
+               f"<ul><li>{'</li><li>'.join(usernames)}</li></ul>"
```

Comment:
This should probably use `html.escape` on each username, but at that point I think this gets a little hard to follow.

What about:

```suggestion
        usernames_html = "".join(f"<li>{html.escape(name)}</li>" for name in usernames)
        text = "Please select an account:<br><ul>{usernames_html}</ul>"
```

(also fixed the "the your" typo)